src/Adapter/PlatformServices.NetCore/Utilities/AssemblyUtility.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.IO;
+    using System.Reflection;
+
+    using Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Deployment;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Utility for assembly specific functionality.
+    /// </summary>
+    internal class AssemblyUtility
+    {
+        private readonly string[] assemblyExtensions = new string[] { ".dll", ".exe" };
+
+        /// <summary>
+        /// Whether file extension is an assembly file extension.
+        /// Returns true for .exe and .dll, otherwise false.
+        /// </summary>
+        /// <param name="extensionWithLeadingDot"> Extension containing leading dot, e.g. ".exe". </param>
+        /// <remarks> Path.GetExtension() returns extension with leading dot. </remarks>
+        /// <returns> True if this is an assembly extension. </returns>
+        internal bool IsAssemblyExtension(string extensionWithLeadingDot)
```

Comment:
>internal [](start = 8, length = 8)

nitpick: public

Reply:
Trying to keep it internal as of now as everything is working correctly with this much access. Will expose publicly if required.